tools/src/Npm.ts

```diff
@@ -75,13 +75,20 @@ export async function publishPackageAsync(
   tagName: string = 'latest',
   dryRun: boolean = false
 ): Promise<void> {
+  // check if two factor auth is required for publishing
+  const npmProfile = await spawnJSONCommandAsync<{
```

Comment:
It might be better if we run this only once and remember that value for subsequent calls. Besides that, could you make a separate function just for getting the profile settings? Also I feel like the `publishPackageAsync` itself should stay atomic and instead let the calling function control the `stdio` as another argument.

Reply:
Sure, makes sense, I've just pushed a commit updating this